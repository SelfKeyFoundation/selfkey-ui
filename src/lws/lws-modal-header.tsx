import * as React from 'react';
import { WithStyles, withStyles, createStyles } from '@material-ui/core';
import { CloseButtonIcon } from '../icons/close-button';
import { SelfkeyLogo } from '../theme/selfkey-dark-theme';
import { Typography } from '@material-ui/core';

const styles = createStyles({
    root: {
        position: 'relative',
        background: '#1F2830',
        borderBottom: '1px solid #43505B',
        borderRadius: '3px 3px 0 0',
        padding: '24px 30px',
        display: 'flex',
        alignItems: 'center'
    },

    closeButton: {
        background: 'none',
        border: 'none',
        margin: 0,
        outline: 'none',
        padding: 0,
        position: 'absolute',
        top: '-18px',
        right: '-18px',
    },

    title: {
        color: '#C5DCE9',
        margin: '0',
        padding: '0 0 0 25px'
    }
});

export type LWSModalHeaderProps = {
  closeAction?: ((event: React.MouseEvent<HTMLElement>) => void),
};


export const LWSModalHeader = withStyles(styles)(({ classes, closeAction }: LWSModalHeaderProps & WithStyles<typeof styles>) => (
    <div className={classes.root}>
        <SelfkeyLogo style={{ width: '50px', height: '43px' }} />
        <Typography variant="h1" className={classes.title}>Login with SelfKey</Typography>
        <button className={classes.closeButton} onClick={closeAction}><CloseButtonIcon/></button>
    </div>
));

export default LWSModalHeader;
