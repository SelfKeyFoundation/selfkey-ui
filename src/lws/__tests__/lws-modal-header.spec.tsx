import * as React from 'react';
import { create } from 'react-test-renderer';
import { LWSModalHeader } from '../lws-modal-header';

describe('LWS Modal Header', () => {
	it('should show LWS Modal Header', () => {
		const component = create(<LWSModalHeader />);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
