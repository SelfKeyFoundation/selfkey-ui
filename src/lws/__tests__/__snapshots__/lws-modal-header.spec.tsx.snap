// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LWS Modal Header should show LWS Modal Header 1`] = `
<div
  className="Component-root-0-1-1"
>
  <svg
    aria-hidden="true"
    className="MuiSvgIcon-root-1 Component-root-0-1-4"
    focusable="false"
    role="presentation"
    viewBox="0 0 53 60"
  >
    <defs>
      <path
        d="M70.1818182,34.4995823 L48.5909091,22 L27,34.4995823 L27,59.5004177 L48.5909091,72 L70.1818182,59.5004177 L70.1818182,34.4995823 Z M55.4868898,32.8086884 C55.4841926,33.5992931 54.8562499,34.2395224 54.0808177,34.2422723 L48.8514749,34.2422723 L48.8514749,37.3299916 L52.7091599,37.3299916 C53.4857121,37.3299917 54.115232,37.9718291 54.115232,38.7635756 C54.115232,39.5553221 53.4857121,40.1971595 52.7091599,40.1971596 L48.8514749,40.1971596 L48.8514749,51.1896408 C48.8539896,51.2402906 48.8539896,51.2910377 48.8514749,51.3416876 C51.5800796,52.0741359 53.3548884,54.7535311 53.0020618,57.6077498 C52.6492352,60.4619686 50.2778561,62.6085761 47.4563291,62.6278292 C44.6348022,62.6470823 42.235455,60.5330285 41.84519,57.6838804 C41.4549251,54.8347322 43.1944,52.1313568 45.9131447,51.3617377 C45.9068271,51.3062722 45.9035445,51.250491 45.9033121,51.1946533 L45.9033121,32.8822055 C45.9033121,32.0517021 46.5636476,31.3784461 47.3782129,31.3784461 C47.5015447,31.3788325 47.6243339,31.395113 47.7436605,31.4269006 C47.859651,31.3951689 47.9791829,31.378882 48.0992755,31.3784461 L54.085734,31.3784461 C54.8611662,31.3811961 55.4891089,32.0214254 55.4918061,32.8120301 L55.4868898,32.8086884 Z"
        id="lws-selfkey-path-1"
      />
      <filter
        filterUnits="objectBoundingBox"
        height="128.6%"
        id="lws-selfkey-filter-2"
        width="132.4%"
        x="-16.2%"
        y="-14.3%"
      >
        <feMorphology
          in="SourceAlpha"
          operator="dilate"
          radius="1.5"
          result="shadowSpreadOuter1"
        />
        <feOffset
          dx="0"
          dy="0"
          in="shadowSpreadOuter1"
          result="shadowOffsetOuter1"
        />
        <feMorphology
          in="SourceAlpha"
          operator="erode"
          radius="1.5"
          result="shadowInner"
        />
        <feOffset
          dx="0"
          dy="0"
          in="shadowInner"
          result="shadowInner"
        />
        <feComposite
          in="shadowOffsetOuter1"
          in2="shadowInner"
          operator="out"
          result="shadowOffsetOuter1"
        />
        <feGaussianBlur
          in="shadowOffsetOuter1"
          result="shadowBlurOuter1"
          stdDeviation="1.5"
        />
        <feColorMatrix
          in="shadowBlurOuter1"
          type="matrix"
          values="0 0 0 0 0.0274509804   0 0 0 0 0.62745098   0 0 0 0 0.694117647  0 0 0 1 0"
        />
      </filter>
    </defs>
    <g
      fill="none"
      fillRule="evenodd"
      id="Modal---Dark"
      stroke="none"
      strokeWidth="1"
    >
      <g
        fillRule="nonzero"
        id="1.-Select-Wallet-Modal"
        transform="translate(-442.000000, -135.000000)"
      >
        <g
          id="Modal-content"
          transform="translate(420.000000, 100.000000)"
        >
          <g
            id="Modal-Title"
            transform="translate(0.000000, 18.000000)"
          >
            <g
              id="title"
            >
              <g
                id="sk-logo"
              >
                <use
                  fill="black"
                  fillOpacity="1"
                  filter="url(#lws-selfkey-filter-2)"
                  xlinkHref="#lws-selfkey-path-1"
                />
                <path
                  d="M48.5909091,21.4222549 L70.6818182,34.2113012 L70.6818182,59.7886988 L48.5909091,72.5777451 L26.5,59.7886988 L26.5,34.2113012 L48.5909091,21.4222549 Z M54.9871835,32.721751 C54.9901385,32.752 54.9917031,32.7826839 54.991809,32.8137358 L55.4918061,32.8120301 L55.7728794,32.3985113 L54.9901014,31.8664484 L54.9871835,32.721751 Z"
                  stroke="#03BED3"
                  strokeWidth="1"
                />
              </g>
            </g>
          </g>
        </g>
      </g>
    </g>
  </svg>
  <h1
    className="Component-base-0-1-5 Component-h1-0-1-6 Component-title-0-1-3"
  >
    Login with SelfKey
  </h1>
  <a
    className="Component-closeButton-0-1-2"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-1 Component-root-0-1-9"
      focusable="false"
      role="presentation"
      viewBox="0 0 38 38"
    >
      <defs />
      <g
        fill="none"
        fillRule="evenodd"
        id="Symbols"
        stroke="none"
        strokeWidth="1"
      >
        <g
          id="Close-button"
          transform="translate(1.000000, 1.000000)"
        >
          <g
            id="Close"
          >
            <circle
              cx="18"
              cy="18"
              fill="#262F39"
              id="Oval-2"
              r="18"
              stroke="#1D505F"
            />
            <g
              fill="#23E6FE"
              fillRule="nonzero"
              id="icons8-delete_sign"
              transform="translate(9.000000, 9.000000)"
            >
              <polygon
                id="Shape"
                points="1.81534091 0.639204545 0.639204545 1.81534091 7.82386364 9 0.639204545 16.1846591 1.81534091 17.3607955 9 10.1761364 16.1846591 17.3607955 17.3607955 16.1846591 10.1761364 9 17.3607955 1.81534091 16.1846591 0.639204545 9 7.82386364"
              />
            </g>
          </g>
        </g>
      </g>
    </svg>
  </a>
</div>
`;
